src/libopenrave/kinbodystatesaver.cpp

```diff
@@ -92,6 +92,10 @@ void KinBody::KinBodyStateSaver::_RestoreKinBody(boost::shared_ptr<KinBody> pbod
         OPENRAVE_ASSERT_OP(pbody->_vGrabbedBodies.size(),==,0);
         for (const GrabbedPtr& pGrabbed : _vGrabbedBodies) {
             KinBodyPtr pGrabbedBody = pGrabbed->_pGrabbedBody.lock();
+            if( !pbody->GetLink(pGrabbed->_pGrabbingLink->GetName()) ) {
```

Comment:
We should do this inside ``if( !!pGrabbedBody ) {``, because ``pGrabbedBody`` could be a null pointer.

Also please check if ``pGrabbed->_pGrabbingLink`` is not null.

Reply:
thank you